### PR TITLE
Service Mesh Updates to AWS Explainer

### PR DESCRIPTION
**Pull request by akundla-google**
Includes a brief section on the AWS Service mesh and updates the load balancer section to note that mesh replaces ILBs when enabled.

### CODE REVIEW

File: bidding_auction_services_aws_guide.md [hunk 1]
```diff
@@ -18,11 +18,11 @@ The FLEDGE Bidding and Auction services will be open-sourced in Q2 2023. In addi
 
 
 #### Comparison to Google Cloud Platform (GCP)
-While this document focuses on AWS, a similar document will be published for GCP in Q2 2023. We intend to provide full GCP and AWS support for these services. Most implementation details will be, at a high level, similar. Two notable exceptions (ultimately with no impact on functionality) are:
+While this document focuses on AWS, a similar document was published for GCP in Q2 2023. We intend to provide full GCP and AWS support for these services. Most implementation details will be, at a high level, similar. A few notable exceptions (ultimately with no impact on functionality) are:
```
Comment (dankocoj-google) [inline bidding_auction_services_aws_guide.md:21]: link to the similar document (preferably the exact section)

File: bidding_auction_services_aws_guide.md [hunk 2]
```diff
@@ -62,9 +62,13 @@ You can create outbound connections from the private subnet using the [NAT gatew
...
+
+#### Service Mesh
+The SellerFrontEnd and BuyerFrontEnd both communicate with the Auction and Bidding services, respectively, using AWS AppMesh, a service mesh. This bypasses any need for an internal load balancer and saves on dedicated load balancer costs, though a flag exists in the terraform configurations to toggle use of service mesh vs internal load balancers. Specifically, the SellerFrontEnd and BuyerFrontEnd use the AWS AppMesh Envoy image to query the AWS AppMesh Envoy Management Service and find routes to available backend services. Each front end relies on its Envoy sidecar to distribute its requests to the appropriate backend service, using a round-robin algorithm amongst healthy instances. Each AppMesh is particular to the region in which it is deployed. Because the requests are internal to the VPC and subnet, they are sent as plaintext gRPC.
+Each service runs a health-checking script on it to support the health checking featues of AWS Cloud Map and AWS Auto-Scaling Group. AWS Cloud Map is a constituent component of the service mesh.
```
Comment (dankocoj-google) [inline bidding_auction_services_aws_guide.md:71]: > Each service runs a health-checking script on it 

'on it' - could you clarify that you mean on the service's EC2 host, but outside of the enclave. 

> health checking featues
typo

> AWS Cloud Map is a constituent component of the service mesh
include a short segment describing what cloud map does

File: bidding_auction_services_aws_guide.md [hunk 3]
```diff
@@ -62,9 +62,13 @@ You can create outbound connections from the private subnet using the [NAT gatew
...
+
+#### Service Mesh
+The SellerFrontEnd and BuyerFrontEnd both communicate with the Auction and Bidding services, respectively, using AWS AppMesh, a service mesh. This bypasses any need for an internal load balancer and saves on dedicated load balancer costs, though a flag exists in the terraform configurations to toggle use of service mesh vs internal load balancers. Specifically, the SellerFrontEnd and BuyerFrontEnd use the AWS AppMesh Envoy image to query the AWS AppMesh Envoy Management Service and find routes to available backend services. Each front end relies on its Envoy sidecar to distribute its requests to the appropriate backend service, using a round-robin algorithm amongst healthy instances. Each AppMesh is particular to the region in which it is deployed. Because the requests are internal to the VPC and subnet, they are sent as plaintext gRPC.
```
Comment (dankocoj-google) [inline bidding_auction_services_aws_guide.md:70]: It'd be good to specify that service mesh is the default (not load balancers)

File: bidding_auction_services_aws_guide.md (hunk 2 above)
Comment (akundla-google) [reply bidding_auction_services_aws_guide.md]: done done and done, thanks

File: bidding_auction_services_aws_guide.md (hunk 3 above)
Comment (akundla-google) [reply bidding_auction_services_aws_guide.md]: done

File: bidding_auction_services_aws_guide.md [hunk 4]
```diff
@@ -18,11 +18,11 @@ The FLEDGE Bidding and Auction services will be open-sourced in Q2 2023. In addi
...
-While this document focuses on AWS, a similar document will be published for GCP in Q2 2023. We intend to provide full GCP and AWS support for these services. Most implementation details will be, at a high level, similar. Two notable exceptions (ultimately with no impact on functionality) are:
+While this document focuses on AWS, a similar document was published for GCP in Q2 2023. We intend to provide full GCP and AWS support for these services. Most implementation details will be, at a high level, similar. A few notable exceptions (ultimately with no impact on functionality) are:
 
 1. GCP is based on a single [virtual machine instance][5], so a parent-child instance relationship will not be required to host the running trusted code in GCP.
 1. The [Envoy component][4] runs inside the Seller Frontend TEE in GCP.
-1. GCP uses a [service mesh][3] to handle its internal load balancing requirements and to allow message routing. AWS uses [load balancers][50]. 
+1. For B&A on AWS, each EC2 runs, outside the TEE: a health-checking script, and an instance of Envoy in a container. These support the [service mesh][3] for AWS. No such infrastructure is needed for mesh on GCP.
```
Comment (salmanmlk) [inline bidding_auction_services_aws_guide.md:25]: Each EC2 instance in AWS runs the following outside TEE: a health-checking script, and an instance of Envoy in a container. These support the [service mesh][3] for AWS. No such infrastructure is needed for mesh on GCP.

File: bidding_auction_services_aws_guide.md [hunk 5]
```diff
@@ -62,9 +62,13 @@ You can create outbound connections from the private subnet using the [NAT gatew
...
+Each front-end service (SFE and BFE) has a load balancer in front. Each load balancer only applies to a particular subnet, so it is regional. The front-end service load balancers are in the public subnet while the Bidding and Auction load balancers, if and when enabled, are in the private subnets. To have a truly global system, it is the ad tech's responsibility to provide a global-scale load balancer that can route requests to its corresponding regional front end load balancers.
 
-The SellerFrontEnd load balancer accepts both HTTP and gRPC traffic over TLS, while all other load balancers take gRPC traffic only. Both of the front end services’ load balancers accept internet traffic, while the load balancers for the Bidding and Auction services are internal-only. Each load balancer has health checks configured for its [target group][19]. A *target group* consists of an autoscaled number of EC2 hosts running a service inside a Nitro Enclave.
```
Comment (salmanmlk) [inline bidding_auction_services_aws_guide.md:67]: HTTP or HTTPS?

File: bidding_auction_services_aws_guide.md (hunk 3 above)
Comment (salmanmlk) [inline bidding_auction_services_aws_guide.md]: nit: "AWS AppMesh Envoy image" or "AWS AppMesh Envoy container"

BTW, this sentence makes it seems like there is some code in SFE / BFE doing this lookup but given this is all transparent to SFE/BFE may be we should reword it to indicate how traffic flows rather than SFE/BFE doing this lookup.

File: bidding_auction_services_aws_guide.md [hunk 6]
```diff
@@ -67,8 +67,8 @@ Each front-end service (SFE and BFE) has a load balancer in front. Each load bal
...
+The SellerFrontEnd and BuyerFrontEnd both communicate with the Auction and Bidding services, respectively, using AWS AppMesh, a service mesh. This bypasses any need for an internal load balancer and saves on dedicated load balancer costs, though a flag exists in the terraform configurations to toggle use of service mesh (the default) vs internal load balancers. Specifically, the SellerFrontEnd and BuyerFrontEnd use the AWS AppMesh Envoy image to query the AWS AppMesh Envoy Management Service and find routes to available backend services. Each front end relies on its Envoy sidecar to distribute its requests to the appropriate backend service, using a round-robin algorithm amongst healthy instances. Each AppMesh is particular to the region in which it is deployed. Because the requests are internal to the VPC and subnet, they are sent as plaintext gRPC.
+Each service runs a health-checking script on the parent EC2 instance (outside the enclave) to support the health checking features of AWS Cloud Map and AWS Auto-Scaling Group. AWS Cloud Map is a constituent component of the service mesh which provides service discovery (tells the mesh which instances are at which IPs).
```
Comment (salmanmlk) [inline bidding_auction_services_aws_guide.md:71]: nit: s/are at which IPs/resolve to what IPs/

**Review by salmanmlk — CHANGES_REQUESTED** [review]
I see 3 commits in the change. Is this expected or should we try to keep one commit here?

File: bidding_auction_services_aws_guide.md (hunk 5 above)
Comment (salmanmlk) [reply bidding_auction_services_aws_guide.md:67]: Makes sense. Thanks.

**Review by salmanmlk — APPROVED** [review]
Thanks for the updates.

**Comment by chatterjee-priyanka** [comment]
Please go ahead and merge. Thanks.
